kyo-core/shared/src/main/scala/kyo/package.scala

```diff
@@ -7,6 +7,13 @@ package object kyo:
 
     object Kyo
 
+    extension [T, S, R](inline kyo: T < (S & Envs[R]))
+        inline def provide(inline layers: Layer[?, ?]*): T < (S & IOs) = {
```

Comment:
Can we move the extension to `kyo-combinators` and provide a `Envs.runLayer`? Maybe we could make `Envs.runTypeMap` not public as well?